h2o-algos/src/main/java/hex/tree/sdt/SDT.java

```diff
@@ -0,0 +1,289 @@
+package hex.tree.sdt;
+
+import hex.ModelBuilder;
+import hex.ModelCategory;
+import org.apache.log4j.Logger;
+import water.DKV;
+import water.exceptions.H2OModelBuilderIllegalArgumentException;
+import water.fvec.Frame;
+import water.util.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Single Decision Tree
+ */
+public class SDT extends ModelBuilder<SDTModel, SDTModel.SDTParameters, SDTModel.SDTOutput> {
+    private int maxDepth;
```

Comment:
We have a project style with leading underscore for Class field naming. So _maxDepth here.

Since its everywhere, we have to follow this style